clients/ui/frontend/src/app/pages/modelCatalog/utils/hardwareConfigurationFilterUtils.ts

```diff
@@ -141,27 +144,45 @@ export const filterHardwareConfigurationArtifacts = (
   });
 
 /**
- * Gets all filter keys (string filters + number filters)
- * TODO: Extend to include latency metric filters when needed for filter chips on details page
+ * Gets all filter keys (string filters + number filters + latency filters)
  */
 export const getAllFilterKeys = (): {
   stringFilterKeys: ModelCatalogStringFilterKey[];
   numberFilterKeys: ModelCatalogNumberFilterKey[];
+  latencyFilterKeys: LatencyMetricFieldName[];
 } => ({
   stringFilterKeys: Object.values(ModelCatalogStringFilterKey),
   numberFilterKeys: Object.values(ModelCatalogNumberFilterKey),
+  latencyFilterKeys: ALL_LATENCY_FIELD_NAMES,
 });
 
 /**
- * Clears all active filters
+ * Clears filters. If filterKeys is provided, only clears those specific filters.
+ * Otherwise clears all filters.
  */
 export const clearAllFilters = (
   setFilterData: <K extends keyof ModelCatalogFilterStates>(
     key: K,
     value: ModelCatalogFilterStates[K],
   ) => void,
+  filterKeys?: ModelCatalogFilterKey[],
 ): void => {
-  const { stringFilterKeys, numberFilterKeys } = getAllFilterKeys();
+  const { stringFilterKeys, numberFilterKeys, latencyFilterKeys } = getAllFilterKeys();
+
+  // If specific filter keys are provided, only clear those
+  if (filterKeys) {
+    filterKeys.forEach((key) => {
+      if (stringFilterKeys.includes(key as ModelCatalogStringFilterKey)) {
+        setFilterData(key as ModelCatalogStringFilterKey, []);
+      } else if (
+        numberFilterKeys.includes(key as ModelCatalogNumberFilterKey) ||
+        isLatencyFieldName(key)
+      ) {
+        setFilterData(key as keyof ModelCatalogFilterStates, undefined);
+      }
+    });
+    return;
```

Comment:
I'm not sure why CI isn't catching the type assertions here.. we also want to avoid these. We can leverage `isEnumMember` imported from `mod-arch-core` here and use a catch-all else case for the non-string filters.


```suggestion
  if (filterKeys) {
    filterKeys.forEach((key) => {
      if (isEnumMember(key, ModelCatalogStringFilterKey)) {
        setFilterData(key, []);
      } else {
        setFilterData(key, undefined);
      }
    });
    return;
```